src/record/record.js

```diff
@@ -151,17 +151,18 @@ Record.prototype.set = function( pathOrData, dataOrCallback, callback ) {
 	var newValue = jsonPath.set( oldValue, path, data, this._options.recordDeepCopy );
 
 	if ( oldValue === newValue ) {
+		if ( typeof callback === 'function' ) utils.setImmediate( callback, null );
```

Comment:
If the connection goes down, calling set will change the data, and buffer the update change to the server.

If the same data is set again, the line you added will return a fake positive.